sunpy/util/functools.py

```diff
@@ -27,5 +27,19 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _singledispatchmethod(func):
+    """
+    A variant of `functools.singledispatch` that can be used on methods.
+    As of python 3.8, this can be replaced by the stdlib `functools.singledispatchmethod` decorator.
+    """
+    dispatcher = functools.singledispatch(func)
+
+    def wrapper(*args, **kw):
+        return dispatcher.dispatch(args[1].__class__)(*args, **kw)
```

Comment:
isn't this actually the same as "seconddispatch", because it's just dispatching on the second arg? The `singledispatchmethod` in py 38 is smarter than this right?

Reply:
Huh, oh yeah... I'll get rid of all this and just use seconddispatch

Reply:
You could conditionally import `singledispatchmethod` if it exists?

Reply:
Can do if you want, although it would make the code a bit messier

Reply:
Oh, but the reason you made the dispatchonsecondarg was for methods right? In that case then yes, I should conditionally import.

Reply:
Just trying the replacement, and it's a bit messy, can I do it in another PR?